src/summary.rs

```diff
@@ -0,0 +1,95 @@
+/// Stores the summary of the farming process into a file.
+/// This allows to retrieve farming information with `info` command,
+/// and also store the amount of potentially farmed blocks during the initial plotting progress,
+/// so that progress bar won't be affected with `println!`, and user will still know about them
+/// when initial plotting is finished.
+use std::path::PathBuf;
+
+use color_eyre::eyre::{Report, Result};
+use serde::{Deserialize, Serialize};
+use tokio::{
+    fs::{read_to_string, File, OpenOptions},
+    io::AsyncWriteExt,
+};
+use tracing::instrument;
+
+#[derive(Deserialize, Serialize, Debug)]
+struct FarmerSummary {
+    initial_plotting_finished: bool,
+    farmed_block_count: u64,
+}
+
+#[instrument]
+pub(crate) async fn create_summary_file() -> Result<()> {
+    let summary_path = summary_path();
+
+    // File::create will truncate the existing file, so first
+    // check if the file exists, if not, `open` will return an error
+    // in this case, create the file
+    // if creation gives an error, that means the directory is not created
+    if File::open(&summary_path).await.is_err() {
+        if File::create(&summary_path).await.is_err() {
+            return Err(Report::msg(
+                "Your config directory is malformed. Please re-run `subspace init`",
```

Comment:
should just keep this error message simple:

```suggestion
                "The config directory does not exist",
```

Including the location of `summary_path` would be good too.